.github/workflows/sim.yaml

```diff
@@ -0,0 +1,111 @@
+name: sim
+
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    types: [opened, synchronize, labeled]
+  schedule:
+    - cron: "0 6 * * *"
+  workflow_dispatch:
+    inputs:
+      make-targets:
+        description: 'Comma separated list of make targets to run (e.g., test-sim-nondeterminism, test-sim-fullappsimulation)'
+        required: true
+        default: 'test-sim-nondeterminism'
+
+concurrency:
+  group: simulation-${{ github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      modified_files: ${{ steps.changes.outputs.modified_files }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Get changed files in x directory
+        id: changes
+        run: |
+          echo "::set-output name=modified_files::$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep '^x/' | xargs)"
+
+  matrix-conditionals:
+    needs: changed-files
+    if: |
+      contains(github.event.pull_request.labels.*.name, 'simulation_tests') || needs.changed-files.outputs.modified_files
```

Comment:
Should we use `SIM_TESTS` label instead?
<img width="306" alt="image" src="https://github.com/user-attachments/assets/e8929719-2fee-4a5a-9a40-889c0a211494">